comp/netflow/server/listener.go

```diff
@@ -12,25 +12,43 @@ import (
 	"github.com/DataDog/datadog-agent/comp/netflow/goflowlib"
 )
 
-// netflowListener contains state of goflow listener and the related netflow config
-// flowState can be of type *utils.StateNetFlow/StateSFlow/StateNFLegacy
 type netflowListener struct {
-	flowState *goflowlib.FlowStateWrapper
-	config    config.ListenerConfig
+	flowState  *goflowlib.FlowStateWrapper
+	config     config.ListenerConfig
+	Error      error
+	errCh      chan error
+	shutdownCh chan struct{}
```

Comment:
Instead of trying to report the error using channels, I think we might be able to use `expvar` also in this case. 

That can make things much simpler I think.

I mean, we can set an expvar value here: https://github.com/DataDog/datadog-agent/blob/1615f92b6a9e81cbd1aa15b3f29e79c3f40096c9/comp/netflow/goflowlib/flowstate.go#L80

And then fetch it when status is requested. 

For example, using expvar like this: https://github.com/DataDog/datadog-agent/blob/36e4383d4f8461e0e2342533dddac0eab0ec25db/pkg/snmp/traps/status.go#L15-L24

In this example, the values are Integer, but it's possible to have String also.